### DISCUSSION TITLE
min-height and height not working correctly

### DISCUSSION THREAD

**Discussion by dantuv90** [tailwindlabs/tailwindcss#12358] [Help] [answered]
When i use tailwind, child div heigth is 0:

```
<div class="min-h-screen">
  <div class="h-full bg-blue-500">
  </div>
</div>
```

Without tailwind it works:

```
<div style="min-height:100vh">
  <div style="height:100%; background-color:blue;">
  </div>
</div>
```

**Comment by Josevictor2** [comment]
up

**Comment by wongjn** [comment]
I wouldn't expect the `height: 100%` to take the height of its parent in either case, and indeed it does not work in either case: https://play.tailwindcss.com/QfxeqAEKgy

It may have been that you had some extra CSS/DOM elements when testing with the `style` version that gave you the false impression of it working.

**Comment by adydetra** [comment]
Check the installation, maybe you missed a [step](https://tailwindcss.com/docs/installation) or see my other [discussions](https://github.com/tailwindlabs/tailwindcss/discussions/12274#discussioncomment-7384298)

[Tailwind Play](https://play.tailwindcss.com/hqoYPJodhn)

**Comment by dantuv90** [comment]
The installation is done correctly. The code with the style doesn't work only on https://play.tailwindcss.com/. Try here https://onecompiler.com/html/3zsfykc9q or create an html by hand and you will see that it works, the child will take all the available height

**Reply by adydetra** [reply]
Didn't include the html structure such as html tags, head, and body? 
onecompiler even provides a template structure when you first create it, but you remove it.

Replace min-h-screen with h-screen

or

If you want to display a blue box, use vh or an exact number like h-40 or h-[5rem], also use width if necessary.

```html
<div class="h-screen">
  <div class="h-full bg-blue-500">
      
  </div>
</div>
  
<div class="h-48">
  <div class="h-full bg-blue-500">
      
  </div>
</div>
  
<div class="h-screen bg-zinc-400">
  <div class="w-20 h-20 bg-red-500">
      
  </div>
</div>
```

**Reply by wongjn** [reply, marked as answer]
Interesting! So it seems like it works when you **don't** include `<!DOCTYPE html>`.
For example, [here is your snippet *not* working with  `<!DOCTYPE html>` added](https://onecompiler.com/html/3zsg2qzge):

```html
<!DOCTYPE html>
<div style="min-height:100vh">
  <div style="height:100%; background-color:blue;"></div>
</div>
```

and [here is the Tailwind version working with no `<!DOCTYPE html>`](
https://onecompiler.com/html/3zsg32wkz):

```html
<div class="min-h-screen">
  <div class="h-full bg-blue-500">
  </div>
</div>
```

Neither option works in Tailwind Play because it automatically adds `<!DOCTYPE html>` to the document.

**Reply by dantuv90** [reply]
Yes, I included them. Same result

**Reply by wongjn** [reply]
> didn't include the html structure such as html tags, head, and body?

@adydetra they would not need to be included – the browser will include them automatically.

**Comment by sandros94** [comment]
So this should be expected? A child element with `h-full` not filling the parent element that has `min-h-screen`?

EDIT: I might be noticing a different problem, in particular when the child element is in a different component from the parent element (vue)

**Comment by theaungmyatmoe** [comment]
Yes the same behaviour is happening in react.

**Comment by wongcheehong** [comment]
Remove `<!DOCTYPE html>` tag in index.html in React somehow make this work again

**Comment by pvautour** [comment]
When removing <!DOCTYPE html> in svelte(kit), firefox gives me this warning:

```
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
```

Removing it may just be forcing the browser to function in a more lenient mode, fixing the issue with tailwind as a side effect. 

I have a feeling a bug would need to be opened as removing doctype is more of a hack/workaround.

Can someone back my reasoning on this?

**Comment by theaungmyatmoe** [comment]
What I found regarding the removal of <!DOCTYPE html> is that it violates web standards.

**Reply by nestorbfgo** [reply]
what did you do ?

**Comment by nestorbfgo** [comment]
min-h-* is not working on React, and the beheavor is same, if I use min-h-40 it doesnt works but if I use something like min-h-[value px] it works . Any update about it?

**Comment by MarcelGDC** [comment]
min-h-* doesn't work to me either, while max-h-* works well :)

**Comment by nestorbfgo** [comment]
Almost I will say all utilities class works but this one not works as expected I don´t. I using Next js  so to solve this I chose min-w-[value]

**Comment by vgoyette** [comment]
Wanted to report this issue as well.
All other utilities I have used work as expected. I tried to use almost every `min-h-*` class to ensure it wasn't just one of them, and none of them compile. However, if I use the "custom" syntax of `min-h-[Nrem]` it works fine.

**Comment by nriin-planit-hhanrichie** [comment]
I had the same issue as well where `min-h-*` doesn't work but `min-h-[value]` works. Apparently the tailwind CSS version has to be `3.4.0` or newer. My tailwind CSS version was `3.3.2` before, after upgrading to `3.4.3` now I can use `min-h-*` & `min-w-*`.
Check here: https://github.com/tailwindlabs/tailwindcss/discussions/12753#discussioncomment-8113495

**Reply by Lexachoc** [reply]
I had this issue as well but I am using Tailwind `3.4.4`.

I can use `min-h-[*]` but not `min-h-*`.

update: I solved the issue by installing tailwindcss again using `npm`, and it just works.... I really don't know what's going on. I'm still running Tailwindcss 3.4.4 though.

**Reply by hdz-666** [reply]
Still facing the issue on 3.4.17

**Comment by Abdelrahmanaman** [comment]
How is this still an issue? I am facing the same issue with tailwind 4.1

**Reply by wongjn** [reply]
It's more a CSS quirk than an intrinsic issue with Tailwind itself. The underlying quirks in CSS that this issue touches upon doesn't change with Tailwind versions.

**Reply by Abdelrahmanaman** [reply]
any workaround to it?

**Reply by wongjn** [reply]
Consider providing a [Tailwind Play](https://play.tailwindcss.com) showcasing your problem, and I could then suggest a remedy.
